### DISCUSSION TITLE
Windows App SDK 1.5.1 (1.5.240311000)

### DISCUSSION THREAD

**Discussion by duncanmacmichael** [microsoft/WindowsAppSDK#4266] [Announcements]
## Windows App SDK 1.5.1 (1.5.240311000)

Windows App SDK 1.5.1 is a stable servicing release that fixes critical bugs for 1.5. 

- If you're building an app with the Windows App SDK for the first time, follow the [installation instructions here](https://docs.microsoft.com/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=desktop-csharp).

- If you are already using the Windows App SDK 1.5 and would like to update to the latest release, you can do so by following the [update instructions here](https://docs.microsoft.com/windows/apps/windows-app-sdk/update-existing-projects-to-the-latest-release).

Check out our [release notes](https://learn.microsoft.com/windows/apps/windows-app-sdk/stable-channel#version-151-15240311000) for the list of issues fixed in this release.

### Updating to this version

The 1.5.1 package is available at https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.5.240311000

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WindowsAppSDK/releases/tag/v1.5.1'>Windows App SDK 1.5.1 (1.5.240311000)</a>.</em>

**Comment by karmeye** [comment]
I still need to add `<UseRidGraph>true</UseRidGraph>` to avoid getting "The specified RuntimeIdentifier 'x' is not recognized" when targeting NET 8. Is that correct?

**Reply by ghost1372** [reply]
Edit all win10 to win
You should update pubxml files too.
I suggest you open your project with vs code. in this way you can search for win10 in all files.

**Reply by kmgallahan** [reply]
> I suggest you open your project with vs code. in this way you can search for win10 in all files.

Just CTRL+Shift+F in VS

**Reply by DarranRowe** [reply]
For a more detailed list of changes, you can find the project templates on the repository [here](https://github.com/microsoft/WindowsAppSDK/tree/main/dev/VSIX/ProjectTemplates/Desktop/CSharp).
Afaik, because of how Visual Studio works, components will only be updated with major changes when Visual Studio has a minor version change. So the next time you should expect to see the project templates to change will be in 17.10.
Yes, the project templates are distributed as part of Visual Studio setup.

**Reply by karmeye** [reply]
Thank you, it seems to work now 👍

**Comment by AlexanderBlackman** [comment]
> Afaik, because of how Visual Studio works, components will only be updated with major changes when Visual Studio has a minor version change. So the next time you should expect to see the project templates to change will be in 17.10.

I just installed VS 17.10 preview but the project templates are still the same.  Creating a project template from github is super difficult for a non-advanced user like me as the test project  you linked too has multiple csprojs that reference things outside the folder.   

It has been years since WinUI3 1.0 so I'm surprised it still hasn't been implemented.  
_Sorry for my negativity, I have problems staying motivated when I hit a snag. Depression +ADHD-RSD_

**Reply by ghost1372** [reply]
https://github.com/winUICommunity/WinUICommunity-Templates

**Comment by AlexanderBlackman** [comment]
> https://github.com/winUICommunity/WinUICommunity-Templates

Thank you, I have your extension, but unfortunately it doesn't have the unit test project introduced in 1.5.
